java/garbage-collector/src/main/java/sleeper/garbagecollector/FailedGarbageCollection.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.garbagecollector;
+
+import sleeper.core.table.TableStatus;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+public class FailedGarbageCollection extends RuntimeException {
```

Comment:
At a glance it's difficult to tell this is an exception. In other places in the code we add an `Exception` suffix to denote this, could we rename this to `FailedGarbageCollectionException`?

Reply:
Done, thanks